### PR TITLE
GEOMESA-2277 Docs - version compatibility

### PR DESCRIPTION
**Pull request by elahrvivaz**
Signed-off-by: Emilio Lahr-Vivaz <elahrvivaz@ccri.com>

### CODE REVIEW

File: docs/user/accumulo/data_management.rst [hunk 1]
```diff
@@ -125,3 +98,89 @@ only populate features matching a CQL filter (e.g. the last month), or choose to
...
+
+        ============= =============== =================================================================
+        Index Version GeoMesa Version Notes
+        ============= =============== =================================================================
+        1             1.0.0           Initial implementation
+        2             1.2.5           Removed redundant feature ID in row value to reduce size on disk
+
+                                      Support for per-attribute visibility
+        3             2.0.0           Standardized index identifier to 'id'
+        ============= =============== =================================================================
```
Comment (jnh5y) [inline docs/user/accumulo/data_management.rst]: This is awesome, @elahrvivaz!

File: docs/user/datastores/data.rst [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+Moving and Migrating Data
+=========================
+
+If you want an offline copy of your data, or you want to move data between networks, you can
+export compressed Avro files containing your simple features. To do this using the command line
+tools, use the export command with the ``format`` and ``gzip`` options:
+
+.. code-block:: bash
+
+    $ geomesa-accumulo export ... -f mySft --format avro --gzip 6 -o myFeatures.avro
```
Comment (jnh5y) [inline docs/user/datastores/data.rst]: Do we have a good way to indicate that 'geomesa-accumulo' is the name of the Accumulo command and that folks can expect to have a similar command for their other datastores?

Comment (elahrvivaz) [reply docs/user/datastores/data.rst]: I'll add a comment

File: docs/user/upgrade.rst [hunk 3]
```diff
@@ -3,10 +3,78 @@
...
-+++++++++++++
+Compatibility Across Versions
++++++++++++++++++++++++++++++
+
+Semantic Versioning
+-------------------
+
+Starting with 2.0.0, GeoMesa is trying to adhere to `semantic versioning <https://semver.org/>`__. Essentially,
+releases are broken down into major, minor and patch versions. For a version number like 2.0.1, 2 is the major
+version, 2.0 is the minor version, and 2.0.1 is the patch version.
```
Comment (jnh5y) [inline docs/user/upgrade.rst]: This is phrased differently than I expected.  I kinda like it...  I was thinking that '0' was the minor version, but that's obviously relative to the Major version 2...

Comment (elahrvivaz) [reply docs/user/upgrade.rst]: yeah, i thought it was clearer that way

**Pull request merged**